src/api/api.jsx

```
@@ -132,20 +143,25 @@ export async function getArticle() {
-  }
-  const query = groq`*[_type == "article" && title match "*${search}*"${hold}] | order(publishedAt desc){}`;
+  try {
+    let hold = "";
+    if(category != undefined && category != "All") {
+      hold = ` && category._ref in *[_type=="articleCategory" && category=="${category}"]._id`;
+    }
+    const query = groq`*[_type == "article" && title match "*${search}*"${hold}] | order(publishedAt desc){}`;
 
-  const articles = await useSanityClient().fetch(query);
-  return articles;
+    const articles = await useSanityClient().fetch(query);
```

Biome (2.3.13)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)